pkgs/top-level/octave-packages.nix

```diff
@@ -0,0 +1,92 @@
+# This file contains the GNU Octave add-on packages set.
+# Each attribute is an Octave library.
+# Expressions for the Octave libraries are supposed to be in `pkgs/development/octave-modules/<name>/default.nix`.
+
+{ pkgs
+, lib
+, stdenv
+, fetchurl
+, newScope
+, octave
+, lapack, blas
+, gfortran
+, autoreconfHook
+, python3
+, python3Packages
+, jdk
+, gnuplot
+, texinfo
+, nettle
+}:
+
+let
+
+  isOctaveFull = octave.enableQt;
+  isOctaveJIT = octave.enableJIT;
+
+  buildOctaveLibrary = callPackage ../development/interpreters/octave/mk-octave-derivation.nix {
+    inherit pkgs lib stdenv;
+    namePrefix = "${octave.pname}-${octave.version}";
+    inherit octave;
+  };
+
+  callPackage = pkgs.newScope {
+    inherit (pkgs) lib stdenv;
+    inherit buildOctaveLibrary;
+    inherit fetchurl;
+    inherit gnuplot texinfo;
+  };
+
+in rec {
+
+  control = callPackage ../development/octave-modules/control {
+    gfortran = gfortran;
+    autoreconfHook = autoreconfHook;
+    lapack = lapack;
+    blas = blas;
+  };
+
+  io = callPackage ../development/octave-modules/io {
+    enableJava = true;
+    jdk = jdk;
```

Comment:
Again, should be the same jdk as octave's.

Reply:
Understood. As mentioned in https://github.com/NixOS/nixpkgs/pull/108562#discussion_r553513390, this has become a moot point for the `io` package. 
However, the `ltfat` package does require a JRE, which I can pass from octave. I will have to handle that later, when I figure out what is causing `ltfat` to have problems building.